src/iosrv/srv.c

```diff
@@ -265,6 +269,28 @@ add_sleep_list(struct dss_xstream *dx, struct dss_sleep_ult *new)
 	d_list_add_tail(&new->dsu_list, &dx->dx_sleep_ult_list);
 }
 
+bool
+dss_xstream_is_busy(void)
+{
+	struct dss_xstream *dx = dss_xstream_get(DSS_XS_SELF);
+	int		    i;
+
+	for (i = 0; i < DSS_POOL_CNT; i++) {
+		size_t cnt;
+		int    rc;
+
+		/* rebuild is a background service, not deemed as "busy" */
+		if (i == DSS_POOL_REBUILD)
+			continue;
+
+		rc = ABT_pool_get_size(&dx->dx_pools[i], &cnt);
```

Comment:
That won't work, we have some background ULTs like GC ULT, also each container has it's own aggregation ULT (they shouldn't affect each other). I suppose we have some statistics for tracking different kinds of ULTs?

Reply:
Indeed, ideally we should have different pools for different operations but I guess this is too performance sensitive and we don't have enough time to verify performance. I will probably just check idle status by counting I/O RPCs.